### DISCUSSION TITLE
Trigger quickadd capture or macro from Android outside of app?

### DISCUSSION THREAD

**Discussion by Fr00t** [chhoumann/quickadd#451] [Ideas]
Would it be possible to trigger a capture or macro from a widget or using Tasker on Android?

Ive set up a simple capture that appends to a section in my daily note, and would love to be able to easily trigger this from my homescreen

Thanks!

**Comment by FelipeRearden** [comment]
The way that I do to trigger commands outside obsidian is to use URL schemes.

There is a plugin called [Obsidian Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri) that creates URL from commands. 

For a QuickAdd command, you will have a URL like this one 

```
obsidian://advanced-uri?vault=XXX&commandid=quickadd%253Achoice%253Ae3e29dd7-a7b7-4c82-af60-a17900bc59a1
``` 

I hope it helps 🙏

**Reply by Fr00t** [reply]
Thats just what I was looking for! Where can I get the choice ID?

Is this documented anywhere? Tried looking but just found the API-docs which I didnt understand

**Reply by FelipeRearden** [reply]
You have to install the plug-in Advanced URI and open command palette to find the commands to create your own URL !!!

Let me know if it worked for you

**Reply by Fr00t** [reply]
Worked perfectly. Set it up as a task in Tasker and added that widget to my homescreen. Thanks a lot!

**Reply by FelipeRearden** [reply]
> Worked perfectly. Set it up as a task in Tasker and added that widget to my homescreen. Thanks a lot!

I am happy to know 😀 

Have a great day ☀️

**Reply by aleksey-rowan** [reply]
> You have to install the plug-in Advanced URI and open command palette to find the commands to create your own URL !!!

If anyone else is confused (like me) and doesn't see the command in the palette, you need to make sure your quickadd is added to the palette by clicking on the lightning bolt toggle in settings:
![image](https://github.com/chhoumann/quickadd/assets/79934725/36f4f6ff-038b-4157-85c3-c6cf86ba0a2c)

There are no tooltips anymore and it's hard to tell what this toggle without them.

**Comment by chhoumann** [comment]
An update on this old one: QuickAdd has its own URI now (since v1.13.0), so the Advanced URI plugin is no longer needed for this:

```
obsidian://quickadd?choice=Your%20Choice%20Name
```

The choice is selected by its name (URL-encoded) - no hunting for choice IDs. You can pass variables as `value-<name>=...` parameters and pick the vault with `vault=`. A Tasker task or homescreen widget that fires that URL intent works exactly like the setup described above.

There's also opt-in x-callback-url support if you want a result back after the choice runs.

Docs: https://quickadd.obsidian.guide/docs/Advanced/ObsidianUri

The Advanced URI approach from this thread still works, but the native URI is simpler. Closing as resolved.
